packages/@aws-cdk/aws-apigateway/lib/authorizers/authorizer-base.ts

```diff
@@ -0,0 +1,91 @@
+import { Construct, Lazy, Resource, ResourceProps } from '@aws-cdk/core';
+import { IAuthorizer } from '../authorizer';
+import { AuthorizationType, Method } from '../method';
+import { IRestApi } from '../restapi';
+
+const AUTHORIZER_BASE_SYMBOL = Symbol.for('@aws-cdk/apigateway.AuthorizerBase');
+
+/**
+ * Base class for all custom authorizers
+ */
+export abstract class AuthorizerBase extends Resource implements IAuthorizer {
+
+  /**
+   * Check if the given object is of type CustomAuthorizer
+   * @internal
+   */
+  public static _isAuthorizer(x: any): x is AuthorizerBase {
+    return x !== null && typeof(x) === 'object' && AUTHORIZER_BASE_SYMBOL in x;
+  }
+
+  /**
+   * The authorizer ID.
+   * @attribute
+   */
+  public abstract readonly authorizerId: string;
```

Comment:
This seems like a duplication, no? Also available in `AuthorizerConfig`...

Reply:
Required since this is a subclass of `IAuthorizer`. Typescript enforces that all abstract subclasses either implement abstract methods and properties, or re-indicate them as being abstract.